metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/HashFunctions.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.stellar.dsl.functions;
+
+import com.google.common.io.BaseEncoding;
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.metron.stellar.dsl.BaseStellarFunction;
+import org.apache.metron.stellar.dsl.Stellar;
+
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+public class HashFunctions {
+  @Stellar(
+    name = "HASH",
+    description = "Hashes a given value using the given hashing algorithm and returns a hex encoded string. This function only hashes " +
+      "strings and values that implement java.io.Serializable.",
+    params = {
+      "toHash - value to hash.",
+      "hashType - A valid string representation of an algorithm supported by java.security.MessageDigest.",
+    },
+    returns = "A hex representation of a hashed value using the given hashing algorithm. If either argument is " +
+      "null then null will be returned. If the type of 'toHash' is neither a string nor of type java.io.Serializable, " +
+      "then null is returned."
+  )
+  public static class Hash extends BaseStellarFunction {
+
+    @Override
+    public Object apply(final List<Object> args) {
+      if (args == null || args.size() != 2) {
+        throw new IllegalArgumentException("Invalid number of arguments: " + (args == null ? 0 : args.size()));
+      }
+
+      final Object toHash = args.get(0);
+      final Object hashType = args.get(1);
+
+      if (toHash == null || hashType == null) {
+        return null;
+      }
+
+      final MessageDigest messageDigest = getMessageDigest(hashType.toString());
+
+      if (toHash instanceof String) {
+        return getHashAsHex(messageDigest, toHash.toString().getBytes(StandardCharsets.UTF_8));
+      } else if (toHash instanceof Serializable) {
+        final byte[] serialized = SerializationUtils.serialize((Serializable) toHash);
+        return getHashAsHex(messageDigest, serialized);
+      }
+
+      return null;
+    }
+
+    private MessageDigest getMessageDigest(String hashType) {
```

Comment:
Never mind, I think I misunderstood you again. You were probably just proposing the code be in a separate directory instead of in the generic stellar/dsl/functions, right?  If so then I'm in left field here.  Prob didn't get enough sleep last night...